code/modules/SCP/scp017.dm

```diff
@@ -0,0 +1,637 @@
+// AI States for SCP-017
+#define IDLE 0
+#define HUNTING 1
+#define STALKING 2
+#define VINDICTIVE 3
+
+/mob/living/simple_animal/hostile/scp017
+	name = "shambling void"
+	desc = "A weird shambling void. You can see nothing inside."
+	icon = 'icons/scp/humanoidscps(32x32).dmi'
+
+	icon_state = "scp-017"
+	response_help_continuous = "tries to reach inside"
+	response_disarm_continuous = "tries to push away"
+	response_harm_continuous = "tries to punch"
+
+	pass_flags = PASSTABLE
+	density = FALSE
+	combat_mode = TRUE //Incase we want to have non-hostile interactions later.
+
+	maxHealth = 100
+	health = 100
+
+	turns_per_move = 1
+	speak_chance = 1
+
+	emote_hear = list("wooshes","whispers")
+	emote_see = list("shambles", "shimmers")
+
+	//Config
+
+	///lumcount required for something to be considered a shadow
+	var/shadow_threshold = 0.35
+
+	// Ability Durations
+	var/stun_duration_light = 20 // 2 seconds
+	var/light_source_debuff_duration = 30 // 3 seconds
+	var/shadow_veil_cooldown = 600 // 60 seconds
+	var/shadow_veil_duration = 150 // 15 seconds
+	var/shadow_sickness_range = 5
+	var/shadow_sickness_duration = 100 // 10 seconds
+	var/void_state_duration = 200 // 20 seconds
+
+	var/last_shadow_veil_time = 0
+	var/list/shadow_sickness_targets = list()
+
+	// Enveloping Shadow and Light Suppression
+	var/enveloping_duration = 30 // 3 seconds for enveloping
+	var/datum/callback/enveloping_progress_timer_id // Timer for the enveloping progress
+	var/mob/living/enveloping_target // The mob currently being enveloped
+
+	var/light_suppression_range = 2
+	var/light_suppression_chance = 10 // 10% chance per tick
+	var/light_suppression_duration = 50 // 5 seconds
+
+	var/list/last_escaped_targets = list() // stores mob/living references and a timestamp
+	var/times_spotted_by_light = 0
+	var/times_driven_off_by_light = 0
+	var/current_ai_state = IDLE // define constants for states
+	var/vindictive_threshold_spotted = 5
+	var/vindictive_threshold_driven_off = 3
+	var/vindictive_duration = 600 // 60 seconds
+	var/last_vindictive_time = 0
+
+	var/list/target_priority_list = list() // Stores mobs with associated priority scores
+
+	var/last_message_time = 0
+	var/message_cooldown = 30 // 3 second cooldown for messages.
+	var/last_stun_time = 0
+	var/stun_cooldown = 50 // 5 second cooldown for re-applying stun
+
+/mob/living/simple_animal/hostile/scp017/Initialize()
+	. = ..()
+	SCP = new /datum/scp(
+		src, // Ref to actual SCP atom
+		"shambling void", //Name (Should not be the scp desg, more like what it can be described as to viewers)
+		SCP_KETER, //Obj Class
+		"017", //Numerical Designation
+	)
+
+//Mob procs
+/mob/living/simple_animal/hostile/scp017/Life(delta_time = SSMOBS_DT, times_fired)
+	. = ..()
+	if(!.) //dead
+		return
+
+	CheckLightExposure()
+	ApplyShadowSickness()
+	HandleLightSuppression() // Call the new light suppression proc
+
+	UpdateTargetPriorities()
+	DecideAction()
+	HandleMovement()
```

Comment:
this stuff really, really should be an AI controller but i am aware that is likely too much to ask at the moment

Reply:
I want to do it; I just don't feel it's very considerate of timeliness right now. I will definitely be revisiting it at some point or another though.